client/src/app/pages/courses/courses.component.ts

```diff
@@ -17,41 +22,36 @@ import { Console } from '@angular/core/src/console';
     public courses: Course[] = [];
     public filteredCourses: Course[] = [];
     public subscription: Subscription;
-    private isLoading: boolean = false;
-
-    constructor(private courseService: CourseService, private searchPipe: SearchPipe) {}
-
-    public ngOnInit() {
-      this.load();
-    }
+    private coursesState: Observable<CoursesState>;
 
-    public load(): void {
-      let query: RequestQuery = { start: 0, count: 10, sort: '' };
-      this.subscription = this.courseService.getAll(query)
+    constructor(private searchPipe: SearchPipe, private store: Store<any>) {
+      this.coursesState = this.store.select((state: any) => state.courses);
+      this.store.select(getCourses)
         .subscribe((courses: Course[]) => {
           this.courses = courses;
           this.filteredCourses = courses;
         });
     }
 
+    public ngOnInit() {
+      this.store.dispatch(new CoursesActions.GetCoursesRequest());
+    }
+
     public onDelete(id: number): void {
       if (confirm('Do you really want to delete this course?')) {
-
-        this.subscription = this.courseService.delete(id)
-          .subscribe((deleted: boolean) => {
-            if (deleted) {
-              console.log(id + ' is deleted');
-              this.load();
-            }
-          });
+        this.store.dispatch(new CoursesActions.DeleteCourseRequest(id));
+        this.store.dispatch(new CoursesActions.GetCoursesRequest());
```

Comment:
these actions are async and can lead to bugs. it's better to get courses after the deletion completes